repository_service_tuf_worker/repository.py

```diff
@@ -831,11 +903,11 @@ def add_targets(
             )
         # The task id will be used by `_send_publish_targets_task` (sub-task).
         task_id = payload.get("task_id")
-        # Group target files by responsible 'bins' delegated roles.
+        # Group target files by responsible delegated role.
         # This will be used to by `_update_task` for updating task status.
         bin_targets: Dict[str, List[targets_models.RSTUFTargetFiles]] = {}
         for target in targets:
-            bins_name = self._get_path_succinct_role(target["path"])
+            delegated_role = self._get_role_for_target_path(target["path"])
```

Comment:
I was testing it, and if you experiment to add a payload with two targets and one of the targets has an invalid target path, it will fail.

```json
{
   "targets":[
      {
         "info":{
            "length":2004,
            "hashes":{
               "sha256":"38744fd5fa3036ad76d4539f98bcef77780bc75216c46e4de9549e90ecdf81f9"
            }
         },
         "path":"files/file3.txt"
      },
      {
         "info":{
            "length":2004,
            "hashes":{
               "sha256":"38744fd5fa3036ad76d4539f98bcef77780bc75216c46e4de9549e90ecdf81f9"
            }
         },
         "path":"file1.txt"
      }
   ]
}
```

The task will fully fail, even only the first one is good.
The problem occurs because `self._get_role_for_target_path` can raise a `StopIteration` (on `next`).

One idea add the same approach as we do in `remove_targets`.
Where we can map artifacts without paths and return in the task result.

As a workaround for this issue, I created a delegated role that has a `delegated_roles` with `path_patterns` as `*`.